src/nextleveldb/Parser.java

```diff
@@ -94,8 +91,9 @@ void documentHandler(String[] tokens) {
 			return;
 		}
 		String name = tokens[2];
-		Document document;
-		switch (operation) {
+		Document document ;
+        document=currentDocument
```

Comment:
No need of duplicate reference here, we can just use `currentDocument` directly wherever we are using `document`